app/Mage.php

```diff
@@ -12,71 +12,95 @@
 
 define('DS', DIRECTORY_SEPARATOR);
 define('PS', PATH_SEPARATOR);
-if (file_exists(__DIR__ . '/vendor/mahocommerce/maho')) {
-    define('BP', dirname(getcwd()));
-} elseif (str_contains(__DIR__, '/vendor/mahocommerce/maho')) {
-    define('BP', str_replace('/vendor/mahocommerce/maho/app', '', __DIR__));
+define('BP', str_replace('/vendor/mahocommerce/maho', '', dirname(__DIR__)));
+
+if (file_exists(BP . '/vendor/mahocommerce/maho')) {
+    define('MAHO_IS_CHILD_PROJECT', true);
+    define('MAHO_FRAMEWORK_DIR', BP . '/vendor/mahocommerce/maho');
 } else {
-    define('BP', dirname(__DIR__));
+    define('MAHO_IS_CHILD_PROJECT', false);
+    define('MAHO_FRAMEWORK_DIR', BP);
 }
 
-Mage::register('original_include_path', get_include_path());
-
 if (!empty($_SERVER['MAGE_IS_DEVELOPER_MODE']) || !empty($_ENV['MAGE_IS_DEVELOPER_MODE'])) {
     Mage::setIsDeveloperMode(true);
     ini_set('display_errors', '1');
     ini_set('error_prepend_string', '<pre>');
     ini_set('error_append_string', '</pre>');
+
+    // Fix for overriding zf1-future during development
+    ini_set('opcache.revalidate_path', 1);
 }
 
+require_once MAHO_FRAMEWORK_DIR . '/app/code/core/Mage/Core/functions.php';
+
+Mage::register('original_include_path', get_include_path());
+
+
 /**
- * Set include path
+ * Require Composer autoloader
  */
-$paths = [];
-$paths[] = BP . '/app/code/local';
-$paths[] = BP . '/app/code/community';
-$paths[] = BP . '/app/code/core';
-$paths[] = BP . '/vendor/mahocommerce/maho/app/code/core';
-$paths[] = BP . '/lib';
-$paths[] = BP . '/vendor/mahocommerce/maho/lib';
-$appPath = implode(PS, $paths);
-set_include_path($appPath . PS . Mage::registry('original_include_path'));
-include_once "Mage/Core/functions.php";
-include_once "Varien/Autoload.php";
+require_once BP . '/vendor/autoload.php';
 
-Varien_Autoload::register();
-require_once BP . DS . 'vendor' . DS . 'autoload.php';
 
-$paths = require BP . '/vendor/composer/include_paths.php';
+/**
+ * Build include paths, note that Composer paths are last
+ */
+$paths = [];
 $paths[] = BP . '/app/code/local';
 $paths[] = BP . '/app/code/community';
-$paths[] = BP . '/app/code/core';
-$allModules = mahoGetComposerInstallationData()[1];
-foreach ($allModules as $module) {
+if (MAHO_IS_CHILD_PROJECT) {
+    $paths[] = BP . '/app/code/core';
+}
+
+$modules = mahoGetComposerInstallationData();
+foreach ($modules as $module => $info) {
     if (str_contains($module, 'mahocommerce/maho')) {
```

Comment:
caus I know there's stuff like `mahocommerce/maho-phpstan-plugin`, but that already shouldn't be picked up because it's filtered out already since it's not `maho-module` or `magento-module`

Reply:
@justinbeaty I think I wanted to remove the core from that foreach

Reply:
We can do `if ($info['type'] === 'maho-source') {` now since that's in the array.

Reply:
we have no control if somebody creates a module and uses type=maho-source so I'd hardcode the project name, seems safer no?

Reply:
If they did do that, then we would just not load it as a module though, just about to push a commit and I'll explain in a comment.

Reply:
mmmm it doesn't seem that we have to limit that in that way when we can check for the project name

Reply:
okay, no problem, so then let's just change to `===` instead of `str_contains` ?

Reply:
absolutely